OsmAnd/src/net/osmand/plus/track/data/SmartFolder.kt

```diff
@@ -23,29 +23,28 @@ class SmartFolder(folderName: String) : TracksGroup, ComparableTracksGroup {
 	@Expose
 	var filters: MutableList<BaseTrackFilter>? = null
 
-	private lateinit var folderAnalysis: TrackFolderAnalysis
+	private var folderAnalysis: TrackFolderAnalysis? = null
 
 	override fun getName(context: Context): String {
 		return folderName
 	}
 
-	override fun getTrackItems(): MutableList<TrackItem> {
+	override fun getTrackItems(): List<TrackItem> {
 		return trackItems
 	}
 
 	fun addTrackItem(trackItem: TrackItem) {
-		trackItems.add(trackItem)
+		if(!trackItems.contains(trackItem)) {
+			trackItems.add(trackItem)
+			folderAnalysis = null
+		}
 	}
 
 	override fun getFolderAnalysis(): TrackFolderAnalysis {
-		if (!this::folderAnalysis.isInitialized) {
+		if (folderAnalysis == null) {
```

Comment:
possible npe if folderAnalysis set to null in resetItems() in another thread, better use local variable